bigquery/samples/query_external_sheets_permanent_table.py

```diff
@@ -0,0 +1,69 @@
+# Copyright 2019 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+def query_external_sheets_permanent_table(dataset_id):
+
+    # [START bigquery_query_external_sheets_perm]
+    from google.cloud import bigquery
+    import google.auth
+
+    # Create credentials with Drive & BigQuery API scopes.
+    # Both APIs must be enabled for your project before running this code.
+    credentials, project = google.auth.default(
+        scopes=[
+            "https://www.googleapis.com/auth/drive",
+            "https://www.googleapis.com/auth/bigquery",
+        ]
+    )
+
+    # TODO(developer): Construct a BigQuery client object.
+    client = bigquery.Client(credentials=credentials, project=project)
+
+    # TODO(developer): Set dataset_id to the ID of the dataset to fetch.
+    # dataset_id = "your-project.your_dataset"
+
+    # Configure the external data source.
+    dataset = client.get_dataset(dataset_id)
+    table_id = "us_states"
+    schema = [
+        bigquery.SchemaField("name", "STRING"),
+        bigquery.SchemaField("post_abbr", "STRING"),
+    ]
+    table = bigquery.Table(dataset.table(table_id), schema=schema)
+    external_config = bigquery.ExternalConfig("GOOGLE_SHEETS")
+    # Use a shareable link or grant viewing access to the email address you
+    # used to authenticate with BigQuery (this example Sheet is public).
+    sheet_url = (
+        "https://docs.google.com/spreadsheets/"
+        "d/1dCG0rrY0nkJpB8t6Ko1S3tY7w9-hp0e_jsYer4LgEuA/edit?usp=sharing"
+    )
+    external_config.source_uris = [sheet_url]
+    external_config.options.skip_leading_rows = 1  # Optionally skip header row.
+    external_config.options.range = (
+        "Sheet1!A10:B30"
```

Comment:
Let's use a range that is more possible to verify the correct behavior. Since the original sheet is sorted alphabetically, the range `A1:B48` should only get "Washington".

Reply:
I decided to use `A20:B49` range. To show that it can be used not from the beginning of the sheet.